### PR TITLE
[frontend] disable mode switching for unmodifiable entity type filter in relationship entities list (#6692)

### LINKED ISSUE #6692: Entity type filters not protected in contextual relationship creation list
## Reproducible Steps
Go to an observable.
Go in the Knowledge tab.
Click on create a relationship.
image

A panel opens with a filter, click on the 'OR' operator between 'Entity' and 'Observable'
image

The operator becomes a 'AND' and an error occurs : 
image

## Expected behavior
No errors and no possibility to switch the operator for this filter in this case

### PR DESCRIPTION
**Pull request by Archidoit**
issue: https://github.com/OpenCTI-Platform/opencti/issues/6692

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntity.tsx [hunk 1]
```diff
@@ -559,7 +559,10 @@ const StixCoreRelationshipCreationFromEntity: FunctionComponent<StixCoreRelation
                   availableEntityTypes={virtualEntityTypes}
                   handleToggleSelectAll="no"
                   entityTypes={virtualEntityTypes}
-                  additionalFilterKeys={{ filterKeys: ['entity_type'], restrictedFiltersConfig: { filterRemoving: ['entity_type'] } }}
+                  additionalFilterKeys={{
+                    filterKeys: ['entity_type'],
+                    restrictedFiltersConfig: { filterRemoving: ['entity_type'], localModeSwitching: ['entity_type'] } }
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntity.tsx]: this is more a general remark : I'm not fond of the names/props used in these parts of the code.

```
additionalFilterKeys={{
  filterKeys: ['entity_type'],
  restrictions: {
    preventRemoveFor: ['entity_type'],
    preventLocalModeSwithcingFor: ['entity_type'],
  }
}}
```

Or something equivalent would be self-explanatory in my opinion.

As for the original bugfix itself, it's ok :)

**Review by labo-flg — APPROVED** [review]

File: opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntity.tsx (hunk 1 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntity.tsx]: fixed

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 2]
```diff
@@ -34,8 +34,8 @@ export type Filter = {
 
 export type RestrictedFiltersConfig = {
   valuesEdition?: string[], // filter keys whose values can't be changed
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: Ideally could you also change this one to `preventEditionFor`  while you're at it? thanks !

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 3]
```diff
@@ -34,8 +34,8 @@ export type Filter = {
 
 export type RestrictedFiltersConfig = {
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: prop restrictedFiltersConfig > filtersRestrictions
type RestrictedFiltersConfig > FiltersRestrictions

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 2 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: done

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 3 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: done :)

**Pull request merged**